.github/workflows/codeql-analysis.yml

```
@@ -29,26 +32,15 @@ jobs:
         uses: github/codeql-action/init@v1
         with:
           languages: ${{ matrix.language }}
+          queries: +security-and-quality
           # If you wish to specify custom queries, you can do so here or in a config file.
           # By default, queries listed here will override any specified in a config file.
           # Prefix the list here with "+" to use these queries and those in the config file.
           # queries: ./path/to/local/query, your-org/your-repo/queries@main
 
-      # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
-      # If this step fails, then you should remove it and run the build manually (see below)
-      - name: Autobuild
-        uses: github/codeql-action/autobuild@v1
-
-      # ℹ️ Command-line programs to run using the OS shell.
-      # 📚 https://git.io/JvXDl
-
-      # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
-      #    and modify them (or add more) to build your code if your project
-      #    uses a compiled language
-
-      #- run: |
-      #   make bootstrap
-      #   make release
+      # Builds the project (only production code without tests)
+      - name: Build Java production code
```

Term production code looks confusing to me, as this is about building artifact. 
```suggestion
      - name: Build Artifact
```

Same the comment about can be updated.

---

That's a good point. In fact we don't have to build the artifact for the scan. I changed it to `./gradlew compileJava` and changed the comment to _Compiles production Java source (without tests)_